include/matx/operators/any.h

```diff
@@ -71,8 +70,17 @@ namespace detail {
       };
 
       template <typename Out, typename Executor>
-      void Exec(Out &&out, Executor &&ex) const {
-        any_impl(cuda::std::get<0>(out), a_, ex);
+      void Exec(Out &&out, Executor) const {
+        auto output_ = cuda::std::get<0>(out);
+        using out_t = decltype(output_);
+        using value_t = typename out_t::value_type;
+        using output_t = typename detail::base_type_t<out_t>;
+
+        output_t out_base = output_;
+        auto op = detail::reduceOpAny<value_t>();
+
+        auto rv = ReduceInputThrust(std::forward<OpA>(a_), std::forward<out_t>(output_), std::forward<decltype(op)>(op));
```

Comment:
We construct the `op`, and then send the inputs and outputs in their original form to the new function `ReduceInputThrust` and of which gets collapsed and dispatched to thrust accordingly.